themes/digital.gov/src/js/glossary.js

```diff
@@ -0,0 +1,251 @@
+ * @param {object} terms json data list of terms and definitions
+ */
+
+function generateListMarkup(terms) {
+  let termsMarkup = "";
+  terms.forEach((term, i) => {
+    const values = {
+      term: term.term,
+      definition: term.definition,
+      definitionClass: cssClasses.definition,
+      itemClass: cssClasses.item,
+      termClass: cssClasses.term,
+      iconClass: cssClasses.icon,
+      termId: i,
+    };
+    termsMarkup += itemTemplate(values);
+  });
+  listElement.insertAdjacentHTML("beforeend", termsMarkup);
+}
+
+/**
+ * Construct new list object from list.js library
+ */
+
+function initializeList() {
+  const listClass = selectors.list.slice(1);
+  const searchClass = selectors.search.slice(1);
+  const options = {
+    valueNames: [cssClasses.term],
+    listClass,
+    searchClass,
+  };
+  list = new List(glossaryBody, options);
+  definitions = glossaryBody.querySelectorAll(`.${cssClasses.term}`);
+}
+
+/**
+ * Hide glossary and disable all focusable elements
+ */
+
+function hideGlossary() {
+  if (toggleButtonTrigger) {
+    glossaryBody.setAttribute("aria-hidden", "true");
+    closeButton.setAttribute("disabled", "");
+    search.setAttribute("disabled", "");
+    definitions.forEach((definition) =>
+      definition.setAttribute("disabled", "")
+    );
+    toggleButtonTrigger.focus({ preventScroll: true });
+    toggleButtonTrigger = null;
+  }
+}
+
+/**
+ * Show glossary and enable all focusable elements
+ * @param {element} toggleButton button which was clicked to open the glossary
+ */
+
+function showGlossary(toggleButton) {
+  if (!toggleButtonTrigger) {
+    glossaryBody.setAttribute("aria-hidden", "false");
+    closeButton.removeAttribute("disabled");
+    search.removeAttribute("disabled");
+    search.focus();
+    definitions.forEach((definition) => definition.removeAttribute("disabled"));
+    toggleButtonTrigger = toggleButton;
+  } else {
+    hideGlossary();
+  }
+}
+
+/**
+ * Toggle glossary term expansion when clicked
+ * @param {event} event fired when term is clicked
+ */
+
+function handleTermClick(e) {
+  if (e.target.matches(`.${cssClasses.term}`)) {
+    const termNumber = e.target.getAttribute("aria-controls");
+    e.target.setAttribute(
+      "aria-expanded",
+      e.target.getAttribute("aria-expanded") === "true" ? "false" : "true"
+    );
+    glossaryBody
+      .querySelectorAll(`#${termNumber}, #${termNumber}-icon`)
+      .forEach((definition) => definition.toggleAttribute("hidden"));
+  }
+}
+
+function initializeEventListeners() {
+  /**
+   * Open glossary when toggle button is clicked
+   */
+  toggleButtons.forEach((button) => {
+    button.addEventListener("click", () => showGlossary(button));
+  });
+
+  /**
+   * Close glossary on escape keypress
+   * @param {event} e to handle keycode lookup when "escape" is pressed
+   */
+  document.body.addEventListener("keyup", (e) => {
+    if (e.keyCode === KEYCODE_ESC) {
```

Comment:
`keyCode` is deprecated, use `.code` instead.

```js
    if (e.code === "Escape") {
```

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode